src/app/pages/my-virtrolio/my-virtrolio.component.ts

```diff
@@ -74,6 +93,42 @@ export class MyVirtrolioComponent implements OnInit {
     }
   }
 
+  shareLink() {
+    if (this.canShare()) {
+      this.navigator.share({
+        title: this.displayName + '\'s virtrolio!',
+        text: 'Visit this link to sign their virtrolio and send them a custom message!',
+        url: this.link,
+      })
+        .then(() => console.log('Successful share'))
+        .catch((error) => console.log('Error sharing', error));
+    } else {
+      alert('Share not supported on this device. Use a mobile device!');
```

Comment:
You should be printing this using `AuthService.displayError` along with the user agent that they were trying to share with so that they can report the error effectively to us.

Reply:
What is a user agent and how do I do that.